packages/connections/src/stores/connections-store.ts

```diff
@@ -351,6 +360,42 @@ export function useConnections(
           type: 'hide-store-connection-error',
         });
       },
+      async saveConnection(connectionInfo: ConnectionInfo) {
```

Comment:
just a personal preference non-blocking at all, this whole object has a lot of code for something that is inside an array in a return statement of another function.

Wouldn't it be better to have these functions defined outside and returned like this: `[state, {fn1, fn2, fn3 ...}]`.

Reply:
In general the main reason React uses this tuple-like return value pattern for its hooks is that they are very generic and allow you to assign the name to the methods without remapping (`var [a, setA] = ...` instead of `var { state: a, setState: a } = ...`), we know exactly what we return here and doesn't look like we need this renaming benefit of having an array, maybe we can just return an object here with `state` and methods?

Reply:
Updated it to an object return, will look into cleaning the function declarations up a bit now